src/sentry/dynamic_sampling/models/common.py

```diff
@@ -0,0 +1,32 @@
+from dataclasses import dataclass
+from typing import Any, List, Optional, Union
+
+import sentry_sdk
+
+from sentry.dynamic_sampling.models.base import Model, ModelInput
+from sentry.dynamic_sampling.rules.utils import ProjectId, TransactionName
+
+
+@dataclass
+class ModelClass:
+    id: Union[ProjectId, TransactionName]
+    count: float
+    new_sample_rate: float = 0.0
+
+
+def sum_classes_counts(classes: List[ModelClass]) -> float:
+    ret_val = 0.0
+
+    for elm in classes:
+        ret_val += elm.count
+
+    return ret_val
+
+
+def guarded_run(model: Model[Any, Any], model_input: ModelInput) -> Optional[Any]:
```

Comment:
nit: looking at the rest of the codebase, it seems like we tend to prefix these this kind of methods with `safe_` rather than `guarded_`

Reply:
Do you have any code examples?